### PR TITLE
Name input

### PR DESCRIPTION
**Pull request by mtwete**
Added the input name screen functionality and also updated run to handle it, new game scores and writing the scores out to the file after the game ends. Unit tests done too. NOTE: You will need to install pygame-textinput first before running!

### CODE REVIEW

File: input_name.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+from pygame_textinput import TextInputVisualizer, TextInputManager
+
+
+class InputName(pygame.sprite.Sprite):
+    #Constructor
+    def __init__(self):
+        super().__init__()
+        #Instruction and re-enter name strings
+        self.instruction_string = "Please input your name(must be between 1 and 15 characters) below:"
+        self.reenter_string = "Your name must be at least 1 character, please try again"
```
Comment (queoui) [inline input_name.py:13]: Always nice to have messages to tell the user what went wrong

File: input_name.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        self.text_input_box = TextInputVisualizer(manager= self.manager, font_object=self.input_text_font,
+                                                  font_color=green_rgb, cursor_color=white_rgb)
+
+        # Single character width, used to calculated background box size behind the text input and other locations
+        self.single_char_width, self.single_char_height = self.input_text_font.size("a")
+        #make the surface height the same as the font and the width the same as the max name length (the just +2 makes the graphics look cleaner)
+        self.input_text_background = pygame.Surface((self.single_char_width * 15 + 2, self.single_char_height))
+        self.input_text_background.fill(white_rgb)
+        #put input box in the center of the screen but offset to the left by half the max length of characters
+        self.image.blit(self.input_text_background, (self.rect.width / 2 - self.single_char_width * (self.max_len / 2), self.rect.height / 2))
```
Comment (queoui) [inline input_name.py:43]: seems like a lot of work to get the format looking good in this section

**Review by queoui — APPROVED** [review]
Thanks Matt for getting this complicated section of the project done. Everything looks clean, I'm looking forward to seeing it in action!

File: input_name.py (hunk 1 above)
Comment (mtwete) [reply input_name.py:13]: Yeah I thought it was a nice touch!

File: input_name.py (hunk 2 above)
Comment (mtwete) [reply input_name.py:43]: It did take some trial and error haha

**Comment by mtwete** [comment]
Thanks I think it looks pretty good too!

**Pull request merged**